src/System.Linq.Parallel/src/System/Linq/Parallel/Utils/ExceptionAggregator.cs

```diff
@@ -116,13 +115,16 @@ internal static void ThrowOCEorAggregateException(Exception ex, CancellationStat
                 CancellationState.ThrowWithStandardMessageIfCanceled(
                     cancellationState.ExternalCancellationToken);
             }
+            else if (ex is AggregateException)
+            {
+                throw ((AggregateException)ex).Flatten();
+            }
```

Comment:
Because it was nesting `AggregateException`s (I can't recall what from now, but it should be one of the immediately surrounding commits).  It makes whatever that method was behave the same as everything else - only a single level of AE before the offending exception.I actually want to investigate moving use of the method in general into other places (with or without this change), since there's similar cancellation checking elsewhere.  Other than that, `QueryTaskState` is doing similar flattening (although the situation isn't completely identical).